### PR TITLE
Removes all wrapped errors in s3_import_dynamodb.go

### PR DESCRIPTION
**Pull request by epps**
# Description

Removes all wrapped errors in s3_import_dynamodb.go

### Select type(s) of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

# Checklist:

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have fixed any merge conflicts

### CODE REVIEW

File: runner/s3_import_dynamodb.go [hunk 1]
```diff
@@ -61,7 +62,7 @@ func (r S3ImportDynamoDBRunner) Run() (types.CompletionWatcher, error) {
 	r.Logger.Infow("Staring S3 import to DynamoDB materialization runner", "name", r.ID.Name, "variant", r.ID.Variant)
 
 	if r.IsUpdate {
-		return nil, fmt.Errorf("materialization updates are not implemented for S3 import to DynamoDB")
+		return nil, fferr.NewInternalError(fmt.Errorf("materialization updates are not implemented for S3 import to DynamoDB"))
```
Comment (sdreyer) [inline runner/s3_import_dynamodb.go:65]: This is because it should be checked before it reaches here right?

File: runner/s3_import_dynamodb.go [hunk 2]
```diff
@@ -78,7 +79,7 @@ func (r S3ImportDynamoDBRunner) Run() (types.CompletionWatcher, error) {
 	sparkOffline, ok := r.Offline.(*provider.SparkOfflineStore)
 	if !ok {
 		r.Logger.Errorf("offline store is not a SparkOfflineStore")
-		return nil, fmt.Errorf("offline store is not a SparkOfflineStore")
+		return nil, fferr.NewInvalidArgument(fmt.Errorf("offline store is not a SparkOfflineStore"))
```
Comment (sdreyer) [inline runner/s3_import_dynamodb.go:82]: maybe "offline store type is not a SparkOfflineStore" ?

File: runner/s3_import_dynamodb.go [hunk 3]
```diff
@@ -99,7 +100,9 @@ func (r S3ImportDynamoDBRunner) Run() (types.CompletionWatcher, error) {
 
 	if len(files) == 0 {
 		r.Logger.Errorf("no files found in source dir path %s", sourceDirPath)
-		return nil, fmt.Errorf("no files found in source dir path %s", sourceDirPath)
+		wrapped := fferr.NewInvalidArgument(fmt.Errorf("no files found in source dir path"))
+		wrapped.AddDetail("sourceDirPath", sourceDirPath.ToURI())
```
Comment (sdreyer) [inline runner/s3_import_dynamodb.go:104]: should be snake case

Comment (sdreyer) [reply runner/s3_import_dynamodb.go:104]: Also should rephrase maybe as "failed to find files in specified directory"

File: runner/s3_import_dynamodb.go [hunk 4]
```diff
@@ -189,37 +192,53 @@ func (w *S3ImportCompletionWatcher) Complete() bool {
...
 	}
 	onlineStore, err := onlineProvider.AsOnlineStore()
 	if err != nil {
-		return nil, fmt.Errorf("failed to convert provider to online store: %v", err)
+		return nil, err
 	}
 	importableOfflineStore, ok := onlineStore.(provider.ImportableOnlineStore)
 	if !ok {
-		return nil, fmt.Errorf("online store is not importable")
+		wrapped := fferr.NewInternalError(fmt.Errorf("online store is not importable"))
```
Comment (sdreyer) [inline runner/s3_import_dynamodb.go:211]: Would change this to specify what the online store is

File: runner/s3_import_dynamodb.go [hunk 5]
```diff
@@ -189,37 +192,53 @@ func (w *S3ImportCompletionWatcher) Complete() bool {
...
+		wrapped.AddDetail("resource_type", runnerConfig.ResourceID.Type.String())
+		return nil, wrapped
 	}
 	offlineStore, err := offlineProvider.AsOfflineStore()
 	if err != nil {
-		return nil, fmt.Errorf("failed to convert provider to offline store: %v", err)
+		return nil, err
 	}
 	if offlineStore.Type() != pt.SparkOffline {
-		return nil, fmt.Errorf("expected offline store to be Spark")
```
Comment (sdreyer) [inline runner/s3_import_dynamodb.go:215]: would add what the given type is

File: runner/s3_import_dynamodb.go [hunk 6]
```diff
@@ -189,37 +192,53 @@ func (w *S3ImportCompletionWatcher) Complete() bool {
...
+		wrapped := fferr.NewInternalError(fmt.Errorf("expected offline store to be Spark"))
+		wrapped.AddDetail("resource_name", runnerConfig.ResourceID.Name)
+		wrapped.AddDetail("resource_variant", runnerConfig.ResourceID.Variant)
+		wrapped.AddDetail("resource_type", runnerConfig.ResourceID.Type.String())
+		return nil, wrapped
 	}
 	sparkOfflineStore, isSparkOfflineStore := offlineStore.(*provider.SparkOfflineStore)
 	if !isSparkOfflineStore {
-		return nil, fmt.Errorf("expected offline store to be Spark")
+		wrapped := fferr.NewInternalError(fmt.Errorf("expected offline store to be Spark"))
```
Comment (sdreyer) [inline runner/s3_import_dynamodb.go:230]: would add the type here as well

File: runner/s3_import_dynamodb.go (hunk 1 above)
Comment (epps) [reply runner/s3_import_dynamodb.go:65]: Are you saying you'd like me to move this check into the factory? I could do that. It actually makes more sense.

**Pull request merged**